docs/guides/tutorials/nextjs_app_router.rst

```diff
@@ -0,0 +1,627 @@
+    
+      return (
+        <div className="container mx-auto p-4 bg-black text-white">
+          <h1 className="text-3xl font-bold mb-4">Posts</h1>
+          <ul>
+            {posts.map((post) => (
+              <li
+                key={post.id}
+                className="mb-4"
+              >
+                <Link
+                  href={`/post/${post.id}`}
+                  className="text-blue-500"
+                >
+                  {post.title}
+                </Link>
+              </li>
+            ))}
+          </ul>
+        </div>
+      )
+    }
+
+Instead of writing our query as a plain string, we're now using the query
+builder to declare our query in a code-first way. As you can see we import the
+query builder as a single default import ``e`` from the ``dbschema/edgeql-js``
+directory.
+
+Now, when we update our ``selectPosts`` query, the type of our dynamically
+loaded ``posts`` variable will update automatically—no need to keep
+our type definitions in sync with our API logic!
+
+Rendering blog posts
+--------------------
+
+Our homepage renders a list of links to each of our blog posts, but we haven't
+implemented the page that actually displays the posts. Let's create a new page
+at ``app/post/[id]/page.tsx``. This is a
+`dynamic route <https://nextjs.org/docs/app/building-your-application/
+routing/dynamic-routes>`_ that
+includes an ``id`` URL parameter. We'll use this parameter to fetch the
+appropriate post from the database.
+
+Add the following code in ``app/post/[id]/page.tsx``:
+
+.. code-block:: tsx
+  
+  import { createClient } from 'edgedb'
+  import e from '@/dbschema/edgeql-js'
+  import Link from 'next/link'
+
+  const client = createClient()
+
+  export default async function Post({ params }: { params: { id: string } }) {
+    const post = await e
+      .select(e.BlogPost, (post) => ({
+        id: true,
+        title: true,
+        content: true,
+        filter_single: e.op(post.id, '=', e.uuid(params.id)),
+      }))
+      .run(client)
+
+    if (!post) {
+      return <div>Post not found</div>
+    }
+
+    return (
+      <div className="container mx-auto p-4 bg-black text-white">
+        <nav>
+          <Link
+            href="/"
+            className="text-blue-500 mb-4 block"
+            replace
+          >
+            Back to list
+          </Link>
+        </nav>
+        <h1 className="text-3xl font-bold mb-4">{post.title}</h1>
+        <p>{post.content}</p>
+      </div>
+    )
+  }
+
+We are again using a Server Component to fetch the post from the database. 
+This time, we're using the ``filter_single`` method to filter the 
+``BlogPost`` type by its ``id``. We're also using the ``uuid`` function 
+from the query builder to convert the ``id`` parameter to a UUID.
+
+Now, click on one of the blog post links on the homepage. This should bring
+you to ``/post/<uuid>``.
+
+Deploying to Vercel
+-------------------
+
+**#1 Deploy EdgeDB**
```

Comment:
I know this was just copied over directly from the other guide, but should we take this opportunity to set up EdgeDB Cloud as the default way to deploy EdgeDB? Then, we can just link to the deployment guides as an alternative for people who want to self-host.

Reply:
I thought about it for a moment, but then I wasn't sure since we don't have a free tier yet. We can mention it, but since it's only a tutorial, EdgeDB could be more costly compared to other providers (+ some have free tiers).

Reply:
I've added the EdgeDB Cloud deployment. There's a bit of repetition, but I figured it might be easier to follow this way (there are "with EdgeDB Cloud" and "with other providers" sections.